### PR TITLE
add features from Spring Data JPA

### PR DESCRIPTION
**Pull request by padu12**
[link to the requirements](https://mate.academy/learn/java-spring-boot/java-spring-boot-data-jpa-v2?section=practice&videoId=5574&testTaskSlug=java_spring_boot_data_jpa)

### CODE REVIEW

**Review by satori11111 — APPROVED** [review]
Incredible! Shocking how you managed to do search in one class

File: src/main/java/com/kaziamyr/onlinebookstore/specification/SpecificationProvider.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.kaziamyr.onlinebookstore.model.Book;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import java.util.List;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SpecificationProvider {
+    public Specification<Book> getSpecification(List<String> params, String fieldName) {
```
Comment (andrii-hoienko) [inline src/main/java/com/kaziamyr/onlinebookstore/specification/SpecificationProvider.java]: Is it possible to make specification provider generic?

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, but have some question

**Review by Vasul-Pankiv — APPROVED** [review]

**Review by vovasalyha — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -20,6 +20,7 @@
         </maven.checkstyle.plugin.configLocation>
         <lombok.mapstruct.binding.version>0.2.0</lombok.mapstruct.binding.version>
         <mapstruct.version>1.5.5.Final</mapstruct.version>
+<!--        <liquibase.version>4.22.0</liquibase.version>-->
```
Comment (boroda4436) [inline pom.xml]: You should not push commented code

File: pom.xml [hunk 3]
```diff
@@ -60,6 +61,16 @@
             <artifactId>mapstruct</artifactId>
             <version>${mapstruct.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>4.22.0</version>
```
Comment (boroda4436) [inline pom.xml]: Why you are not using `liquibase.version` property here?

File: src/main/java/com/kaziamyr/onlinebookstore/controller/BookController.java [hunk 4]
```diff
@@ -24,12 +30,29 @@ public List<BookDto> findAll() {
...
     }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("/{id}")
+    public void deleteById(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public BookDto updateById(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
```
Comment (boroda4436) [inline src/main/java/com/kaziamyr/onlinebookstore/controller/BookController.java]: could be just `update`
```suggestion
    public BookDto update(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
```

File: src/main/java/com/kaziamyr/onlinebookstore/model/Book.java [hunk 5]
```diff
@@ -8,8 +8,12 @@
 import jakarta.persistence.Table;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
```
Comment (boroda4436) [inline src/main/java/com/kaziamyr/onlinebookstore/model/Book.java]:
```suggestion
@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id = ?")
```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-books-table.yaml
+  - include:
+      file: db/changelog/changes/02-alter-books-table.yaml
```
Comment (boroda4436) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: changeset file name should describe what changes were applied. The `02-alter-books-table` is too generic

**Pull request merged**